app/models/utility/cell_count_spot_checking.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Utility
+  # This class is used to select wells from a plate for spot checking.
+  class CellCountSpotChecking
+    attr_reader :plate, :ancestor_tubes
+
+    # Initializes the object with a plate and its ancestor tubes. It is used
+    # to pick wells from the plate for spot checking.
+    #
+    # @param plate [Plate] the plate associated with this instance
+    # @param ancestor_tubes [Array<Tube>] the ancestor tubes of the plate
+    def initialize(plate, ancestor_tubes)
+      @plate = plate
+      @ancestor_tubes = ancestor_tubes
```

Comment:
You shouldn't have to retrieve the ancestor tubes, because the Vac tube barcode is saved in the `sample_metadata` `supplier_name`. (specified in docs so should be consistent https://ssg-confluence.internal.sanger.ac.uk/display/PSDPUB/scRNA+Core+Pipeline+Documentation)

Reply:
As requested, I have updated several places now so it relies on sample_metadata supplier name for Vac tube barcodes, instead of ancestor_tubes via sample uuid access. Also, ancestor_tubes has been removed from export includes to stop retrieving them.